drivers: dac: Add driver support for TI MSPM0 G-Series DAC module

santhosh-c-c opened the pull request:
**Add driver support for TI MSPM0 G-Series DAC module**

Add support for the DAC module on TI’s MSPM0 G-Series MCUs. The DAC
supports 8-bit and 12-bit resolution, configurable via driver APIs.

The driver also supports selecting the voltage reference source for the DAC output.
By default, the internal reference is used, but an external voltage reference
can be selected via Kconfig options.

**Signed-off-by:**
Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
dts: arm: ti: mspm0: g: add DAC nodes for TI MSPM0 G-Series

Add DAC devicetree nodes for MSPM0 G-Series SoCs that include the DAC
peripheral: mspm0g150x, mspm0g350x and mspm0gx51x.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
drivers: dac: add driver support for TI MSPM0 G-Series DAC module

Add support for the DAC module on TI’s MSPM0 G-Series MCUs. The DAC
supports 8-bit and 12-bit resolution.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
dts: bindings: dac: add DT bindings for TI MSPM0 G-Series DAC module

Add devicetree bindings for the TI MSPM0 G-Series DAC module.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

parthitce commented on drivers/dac/Kconfig.mspm0:
```diff
@@ -0,0 +1,28 @@
+	  Enable support for the DAC (Digital-to-Analog Converter)
+	  peripheral driver for Texas Instruments MSPM0 G-Series MCUs.
+
+config DAC_MSPM0_TIMEOUT_MS
```
Am not sure if this needs to be from Kconfig.

martinjaeger reviewed (changes requested):
Can you please add support for this driver to at least one board and to the `dac_api` test to ensure it is built in CI?

Also please fix remaining compliance issues raised by CI.

martinjaeger commented on samples/drivers/dac/boards/lp_mspm0g3507.overlay:
```diff
@@ -0,0 +1,17 @@
+};
+
+&dac0 {
+	status = "okay";
```
This should be done on borad level. And the `dac` should also be added to the supported features in the board's .yaml file and documentation.

santhosh-c-c committed:
boards: ti: lp_mspm0g3507: enable DAC support

Enable DAC peripheral and add dac to the list of supported
hardware in board YAML.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
boards: ti: lp_mspm0g3519: enable DAC support

Enable DAC peripheral and add dac to the list of supported
hardware in board YAML.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

martinjaeger reviewed (dismissed):
Looks good. One minor remark which is not blocking.

santhosh-c-c committed:
tests: drivers: dac: dac_api: add support for MSPM0G series

Add lp_mspm0g3507 and lp_mspm0g3519 to the dac0_channel0 test
configuration in testcase.yaml.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

ssekar15 commented:
@santhosh-c-c fix the compliance issue
